### PR TITLE
feat(lib): 🎸 add messageformat support through custom transpiler (#11)

### PR DESCRIPTION
**Pull request by Coly010**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngneat/transloco/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

The library does not currently support `messageformat`. 

Issue Number: #11 

## What is the new behavior?

Add a custom transpiler: `MessageFormatTranspiler` to handle `messageformat` formatted translations.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### Usage

To enable support within Transloco app the following to the providers array in your `app.module.ts`:

```ts
import { MessageFormatTranspiler } from '@ngneat/transloco';

...

@NgModule({
  providers: [
    ...,
    { provide: TRANSLOCO_TRANSPILER, useClass: MessageFormatTranspiler }
  ]
})

```

The `MessageFormatTranspiler` is compatible with the `DefaultTranspiler` and therefore you can switch without worry that it will break your current translations.

It then enables support for the following within your i18n translation files:

```js
{
  "mySelectRule": "{myVar, select, val1 {Value 1} val2 {Value 2} other {Other Value}}",
  "myPluralRule": "{myCount, plural, =0 {no results} one {1 result} other {# results}}"
}
```

### CODE REVIEW

**Comment by shaharkazaz** [comment]
⁦❤️⁩

File: projects/ngneat/transloco/src/lib/transpiler-strategies/messageformat.transpiler.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { TranslocoTranspiler, DefaultTranspiler } from '../transloco.transpiler';
+import { isString } from '../helpers';
+import { HashMap, Translation } from '../types';
+
+import MessageFormat from 'messageformat';
+
+export class MessageFormatTranspiler implements TranslocoTranspiler {
+  transpile(value: string, params: HashMap = {}, translation: Translation): string {
+    const defaultTranspiler = new DefaultTranspiler();
```
Comment (shaharkazaz) [inline projects/ngneat/transloco/src/lib/transpiler-strategies/messageformat.transpiler.ts]: new DefaultTranspiler() and new MessageFormat() can be instantiated once

File: src/app/app.module.ts [hunk 2]
```diff
@@ -23,10 +29,12 @@ import { webpackLoader } from './loaders/webpack.loader';
       useValue: {
         prodMode: environment.production,
         listenToLangChange: true,
-        fallbackLang: 'es',
+        fallbackLang: 'en',
```
Comment (shaharkazaz) [inline src/app/app.module.ts]: why changed? 🤔

File: src/app/transpilers/transpilers.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (shaharkazaz) [inline src/app/transpilers/transpilers.component.spec.ts]: It's not really testing anything, please remove it

**Comment by shaharkazaz** [comment]
@Coly010 Awesome job man! enjoyed going over this 🙂

File: src/app/app.module.ts (hunk 2 above)
Comment (Coly010) [reply src/app/app.module.ts]: My mistake. It's leftovers from debugging. Will remove.

File: projects/ngneat/transloco/package.json [hunk 4]
```diff
@@ -42,5 +42,8 @@
     "replace-in-file": "^4.1.2",
     "ora": "^3.4.0"
   },
+  "peerDependencies": {
```
[the remark this answers is not in the document]
Comment (Coly010) [reply projects/ngneat/transloco/package.json]: Ok. I'll update the docs to direct users to install it if they decide to use.

**Pull request merged**